lib/reply.js

```diff
@@ -65,31 +65,38 @@ function Reply (res, context, request, log) {
   this.log = log
 }
 
-Object.defineProperty(Reply.prototype, 'sent', {
-  enumerable: true,
-  get () {
-    return this[kReplySent]
-  },
-  set (value) {
-    if (value !== true) {
-      throw new FST_ERR_REP_SENT_VALUE()
-    }
-
-    if (this[kReplySent]) {
-      throw new FST_ERR_REP_ALREADY_SENT()
+Object.defineProperties(Reply.prototype, {
+  res: {
+    get () {
+      process.emitWarning('You are accessing the Node.js core response object via "reply.res", Use "reply.raw" instead.')
```

Comment:
Can/should we follow this recommended practice? https://nodejs.org/dist/latest-v12.x/docs/api/process.html#process_avoiding_duplicate_warnings 

Also, it may be good to:

```js
const warning = Error('the message')
warning.name = 'FastifyDeprecation'
warning.code = 'FSTDEP001'
```

Then we should add a deprecations document like https://nodejs.org/dist/latest-v12.x/docs/api/deprecations.html to record what we have deprecated.

Reply:
Yes, this is useful because it's not straightforward to understand who emitted the warning.
Regarding the deprecations document, it might be overkill for now, if we will start issuing more deprecations in the future, we'll add it.